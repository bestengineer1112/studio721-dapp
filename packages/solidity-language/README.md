# solidity-language

A close-enough solidity AST
