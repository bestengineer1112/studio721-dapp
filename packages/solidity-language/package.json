{
  "name": "solidity-language",
  "version": "1.0.0",
  "main": "src/index.ts",
  "license": "UNLICENSED",
  "files": [
    "README.md"
  ]
}
