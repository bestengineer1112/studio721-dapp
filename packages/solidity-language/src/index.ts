export * as AST from './ast';
export * from './format';
export * from './builders';
