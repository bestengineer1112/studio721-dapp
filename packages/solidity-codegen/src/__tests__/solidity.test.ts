import * as s from 'solidity-language';

describe('format', () => {
  it('literal', () => {
    const node = s.formatLiteral({
      type: 'numberLiteral',
      value: '123',
    });

    expect(s.print(node)).toMatchSnapshot();
  });

  it('variable declaration', () => {
    const node = s.formatVariableDeclaration({
      type: 'variableDeclaration',
      name: 'foo',
      typeAnnotation: 'uint256',
      modifiers: ['public'],
      initializer: {
        type: 'literal',
        value: { type: 'numberLiteral', value: '123' },
      },
    });

    expect(s.print(node)).toMatchSnapshot();
  });

  it('contract', () => {
    const node = s.formatContract({
      type: 'contractDeclaration',
      name: 'TestToken',
      extends: ['ERC721', 'ReentrancyGuard', 'Ownable'],
      body: [
        {
          type: 'variableDeclaration',
          name: 'foo',
          typeAnnotation: 'uint256',
          modifiers: ['public'],
          initializer: {
            type: 'literal',
            value: { type: 'numberLiteral', value: '123' },
          },
        },
        {
          type: 'variableDeclaration',
          name: 'bar',
          typeAnnotation: 'uint256',
          modifiers: ['public'],
          initializer: {
            type: 'literal',
            value: { type: 'numberLiteral', value: '123' },
          },
        },
      ],
    });

    expect(s.print(node)).toMatchSnapshot();
  });

  it('import', () => {
    const node = s.formatImport({
      path: '@abc',
      names: [],
    });

    expect(s.print(node)).toMatchSnapshot();
  });

  it('pragma', () => {
    const node = s.formatPragma({
      value: 'solidity ^0.8.9',
    });

    expect(s.print(node)).toMatchSnapshot();
  });

  it('program', () => {
    const node = s.formatProgram({
      license: 'MIT',
      pragma: { value: 'solidity ^0.8.9' },
      imports: [
        { path: '@a', names: [] },
        { path: '@b', names: [] },
        { path: '@c', names: [] },
      ],
      body: [],
    });

    expect(s.print(node)).toMatchSnapshot();
  });
});
