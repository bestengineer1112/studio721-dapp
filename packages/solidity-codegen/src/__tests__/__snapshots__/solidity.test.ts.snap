// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format contract 1`] = `
"contract TestToken is ERC721, ReentrancyGuard, Ownable {
  uint256 public foo = 123;

  uint256 public bar = 123;
}"
`;

exports[`format import 1`] = `"import \\"@abc\\";"`;

exports[`format literal 1`] = `"123"`;

exports[`format pragma 1`] = `"pragma solidity ^0.8.9;"`;

exports[`format program 1`] = `
"// SPDX-License-Identifier: MIT

pragma solidity ^0.8.9;

import \\"@a\\";
import \\"@b\\";
import \\"@c\\";"
`;

exports[`format variable declaration 1`] = `"uint256 public foo = 123;"`;
